crypto/kzg/src/lib.rs

```diff
@@ -29,42 +43,34 @@ impl From<c_kzg::Error> for Error {
     }
 }
 
-pub const CELLS_PER_EXT_BLOB: usize = 128;
-
-// TODO(das): use proper crypto once ckzg merges das branch
-#[allow(dead_code)]
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
-pub struct Cell {
-    bytes: [u8; 2048usize],
-}
-
-impl Cell {
-    pub fn from_bytes(b: &[u8]) -> Result<Self, Error> {
-        Ok(Self {
-            bytes: b
-                .try_into()
-                .map_err(|_| Error::Kzg(c_kzg::Error::MismatchLength("".to_owned())))?,
-        })
-    }
-    pub fn into_inner(self) -> [u8; 2048usize] {
-        self.bytes
-    }
-}
-
 /// A wrapper over a kzg library that holds the trusted setup parameters.
 #[derive(Debug)]
 pub struct Kzg {
     trusted_setup: KzgSettings,
+    context: DASContext,
 }
 
 impl Kzg {
     /// Load the kzg trusted setup parameters from a vec of G1 and G2 points.
     pub fn new_from_trusted_setup(trusted_setup: TrustedSetup) -> Result<Self, Error> {
+        // Initialize the trusted setup using default parameters
+        //
+        // Note: One can also use `from_json` to initialize it from the consensus-specs
+        // json string.
+        let peerdas_trusted_setup = PeerDASTrustedSetup::default();
```

Comment:
Yep that's right. Yeah I think equivalence check is a good idea, or potentially cloning the values from existing trusted setup. There are a few more things I've done on another WIP branch, but I'm still thinking about whether it's a good idea:
- Should we only load this if PeerDAS is scheduled (`None` otherwise), so we don't impact mainnet code path now (I have a branch [here](https://github.com/sigp/lighthouse/blob/89fd62a3c08c0e099099a0883c703e01943d55dc/crypto/kzg/src/lib.rs#L61-L73) that does this)
- Load PeerDAS setup from existing `trusted_setup` using clone ([code](https://github.com/sigp/lighthouse/blob/89fd62a3c08c0e099099a0883c703e01943d55dc/crypto/kzg/src/lib.rs#L66)), so we avoid reading file system and also avoid breaking `--trusted-setup` - although we don't really use this anymore

Reply:
I haven't done #1, it actually involves touching a lot of files and I'm not sure if it's worth it, as loading the PeerDAS trusted setup only adds ~550ms on startup and ~100mb of memory for storing the precomputation. Existing Deneb code path still uses c-kzg so should be unaffected.

What do you think @realbigsean ?

Reply:
Replaced loading embedded trusted setup with cloning the existing trusted setup values.
https://github.com/sigp/lighthouse/pull/6212/commits/11c53a81b915eef3b0a73cec657c5bc177f6db6c

Reply:
Added conditionally load KZG on startup only if PeerDAS is enabled: https://github.com/sigp/lighthouse/pull/6212/commits/688198155d9c2fa030710a0d9b5e1f029b84f8f5

Reply:
> Once both libraries are feature complete and stable for deneb and PeerDAS, we could do this properly and only load one library with a cli flag.

That sounds good!